### DISCUSSION TITLE
How to load relation of a relation?

### DISCUSSION THREAD

**Discussion by ninode97** [prisma/prisma#11143] [Q&A] [answered]
Hello, I have picked prisma2 as ORM for my new project.

I run into one consideration, how could I load relation of relation? 

For example I have such table definitions:

```
model Brand {
  id      Int     @id @default(autoincrement())
  name    String  @unique
  addedBy User    @relation(fields: [userId], references: [id])
  userId  Int
  models  Model[]
}

model Model {
  id      Int    @id @default(autoincrement())
  name    String
  addedBy User   @relation(fields: [userId], references: [id])
  userId  Int
  Brand   Brand? @relation(fields: [brandId], references: [id])
  brandId Int?
  Car     Car[]
}

model Car {
  id                           Int      @id @default(autoincrement())
  plateCode                    String
  vinCode                      String   @unique
  addedBy                      User     @relation(fields: [userId], references: [id])
  userId                       Int
  ownedByCompany               Company  @relation(fields: [companyId], references: [id])
  companyId                    Int
  model                        Model    @relation(fields: [modelId], references: [id])
  modelId                      Int
  year                         Int
  acquiredDate                 DateTime
  insuranceValidFrom           DateTime
  insuranceExpiresOn           DateTime
  technicalInspectionValidFrom DateTime
  technicalInspectionExpiresOn DateTime
}
```

When I want to retrieve my data I am getting this response:
```
[
    {
        "id": 1,
        "plateCode": "XXQ:000",
        "vinCode": "ZXQQQQQQQQQQQQQQQQQQQ",
        "userId": 1,
        "companyId": 1,
        "modelId": 1,
        "year": 2004,
        "acquiredDate": "2021-12-15T21:02:57.206Z",
        "insuranceValidFrom": "2020-02-01T21:02:57.206Z",
        "insuranceExpiresOn": "2023-02-01T21:02:57.206Z",
        "technicalInspectionValidFrom": "2021-06-25T21:02:57.206Z",
        "technicalInspectionExpiresOn": "2023-06-25T21:02:57.206Z",
        "model": {
            "id": 1,
            "name": "525i",
            "userId": 1,
            "brandId": 1
        }
    }
]
```

Is there a way to populate **brandId** in similar fashion with **include** object?
Or maybe I could define virtual / computed fields within schema definition? 

I know that I could transform / populate that field with second call to db but just thinking whether there is a convenient way to do it.
Any help would be appreciated. :)

**Comment by ninode97** [comment, marked as answer]
I have posted same question in stackoverflow and received answer that solves this thing for me. 

```
  prisma.car.findMany({
    include: {
      model: {
        include: {
          Brand: true,
        },
      },
    },
  });
```
Didn't knew that you guys implemented such simple approach.

I really start to love prisma ORM!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
